apps/src/music/views/MusicView.jsx

```diff
@@ -663,10 +665,8 @@ class UnconnectedMusicView extends React.Component {
 
     this.sequencer.clear();
     this.musicBlocklyWorkspace.executeCompiledSong(this.playingTriggers);
-    this.props.addPlaybackEvents({
-      events: this.sequencer.getPlaybackEvents(),
-      lastMeasure: this.sequencer.getLastMeasure(),
-    });
+    this.props.addPlaybackEvents(this.sequencer.getPlaybackEvents());
+    this.props.setLastMeasure(this.sequencer.getLastMeasure());
```

Comment:
Should we use `updateLastMeasure` here too in case this executes while playback is ongoing, and the new set of when_run events is shorter than an ongoing trigger?

Reply:
Oh the good news is that even if the `when_run` events are shortened while playback is ongoing, `this.sequencer.getLastMeasure()` still returns the measure at the end of the longer ongoing triggered sounds.